nodejs/awsx/autoscaling/autoscaling.ts

```diff
@@ -60,24 +62,37 @@ export class AutoScalingGroup extends pulumi.ComponentResource {
         this.targetGroups = args.targetGroups || [];
         const targetGroupArns = this.targetGroups.map(g => g.targetGroup.arn);
 
+        let launchConfigurationId: pulumi.Input<string>;
+        let stackName: pulumi.Input<string>;
+
         // Use the autoscaling config provided, otherwise just create a default one for this cluster.
-        if (args.launchConfiguration) {
-            this.launchConfiguration = args.launchConfiguration;
+        if (args.launchConfigurationId) {
+            launchConfigurationId = args.launchConfigurationId;
+            stackName = new random.RandomId(name, {
+                byteLength: 8,
+                prefix: name,
+                keepers: { launchConfigurationId },
+            }, { parent: this }).b64Url;
```

Comment:
not sure if this is the right way to do things.  But currently we use a random bucket id to get a unique stack name.  i figured we should likely switch to actually pulumi.Random for something cleaner.